core/app/models/concerns/spree/user_address_book.rb

```diff
@@ -0,0 +1,73 @@
+module Spree
+  module UserAddressBook
+    extend ActiveSupport::Concern
+
+    included do
+      has_many :user_addresses, foreign_key: "user_id", class_name: "Spree::UserAddress"
+
+      has_many :addresses, through: :user_addresses
+
+      has_one :default_user_address, -> { where default: true}, foreign_key: "user_id", class_name: 'Spree::UserAddress'
+      has_one :default_address, through: :default_user_address, source: :address
+
+      # bill_address is only minimally used now, but we can't get rid of it without a major version release
```

Comment:
Is the long term plan to continue supporting the existing data model of a user having a bill address, or is the plan to move that association to the credit card model and only have a default/shipping address on the user?